src/conn/conn_sweep.c

```diff
@@ -359,8 +366,14 @@ __wt_sweep_create(WT_SESSION_IMPL *session)
 	/*
 	 * Handle sweep does enough I/O it may be called upon to perform slow
 	 * operations for the block manager.
+	 *
+	 * The sweep thread sweeps the lookaside table for outdated records,
+	 * it gets its own cursor for that purpose.
+	 *
+	 * Don't tap the sweep thread for eviction.
 	 */
-	F_SET(session, WT_SESSION_CAN_WAIT);
+	F_SET(session, WT_SESSION_CAN_WAIT |
+	    WT_SESSION_LOOKASIDE_CURSOR | WT_SESSION_NO_EVICTION);
```

Comment:
What do you mean by "at the top level"?

I think of LRU eviction as exactly the "opportunistic" case you are talking about -- threads other than the eviction workers choose a page from the LRU queue to evict if they can't make progress when the cache is full.

Or we could flip this around, and mark ordinary application sessions with `WT_SESSION_CAN_EVICT`?